src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/SecretUtils.java

```diff
@@ -104,15 +104,21 @@ public static String base64DecodeToString(String s) {
 
     /**
      * Get the scope from a given {@code Secret}.
+     * If the label is not supported or empty, then it defaults to global scope.
      * @param s the secret whose scope we want to obtain.
      * @return the scope for a given secret.
      */
     public static CredentialsScope getCredentialScope(Secret s) {
-        CredentialsScope scope = CredentialsScope.valueOf(Optional
-                .ofNullable(s.getMetadata().getLabels().get(JENKINS_IO_CREDENTIALS_SCOPE_LABEL))
-                .orElse("global")
-                .toUpperCase(Locale.ROOT));
-        LOG.log(Level.FINE, "Scope for secret {0} is set to {1}", new Object[] {getCredentialId(s), scope});
+        CredentialsScope scope = CredentialsScope.GLOBAL;
+        String label = s.getMetadata().getLabels().get(JENKINS_IO_CREDENTIALS_SCOPE_LABEL);
+        if (label != null) {
+            try {
+                scope = CredentialsScope.valueOf(label.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException exception) {
+                LOG.log(Level.FINE, "Secret {0} contains an unsupported scope label {0} - defaults to global scope",
+                        new Object[]{getCredentialId(s), label});
```

Comment:
I am actually in 2 minds if this is the correct way to handle something.  it may let a SYSTEM scope secret actually become GLOBAL with nothing except a log to warn you.  the secret is invalid - so not parsing it is perfectly valid.

```suggestion
                throw new CredentialsConvertionException(JENKINS_IO_CREDENTIALS_SCOPE_LABEL + " is set to an invalid scope: " + label, exception);
```

Reply:
My opinion is I would rather have the exception thrown than the secret accidentally (and possibly unknowingly) exposed to users.

Reply:
Thanks for your efforts here @fowlie  if you make these changes I will try and get a release out this week.

Reply:
My pleasure! :smile: Hm, the last build there failed (linux agent was marked offline). How to start another build....?

Reply:
The checks tab should have a " re run" option

Reply:
> The checks tab should have a " re run" option

I can't find it, maybe I don't have permission?